keps/sig-node/3815-pod-condition-for-configuration-errors/README.md

```diff
@@ -0,0 +1,1069 @@
+bogged down.
+-->
+
+#### Story 1
+
+As an end-user, if I create a pod with an invalid image, I would expect to see a condition that reflects that the pod failed to start due to a configuration error.
+
+#### Story 2
+
+As an end-user, if I create a pod with an ImagePullPolicy of Never but image doesn't exist on the node, I would expect to see a condition that states that the pod failed to start due to a configuration error.  
+
+#### Story 3
+
+As an end-user, if I create a pod with a missing config map, I would expect to see a condition that states that the pod failed to start due to a configuration error.
+
+### Notes/Constraints/Caveats (Optional)
+
+#### Pending Pod State Review
+
+##### Invalid Image Name
+
+- Reproduction: Run image with invalid image name (all capitals)
+- Pod status:
+  - status: Pending
+- ContainerStatus:
+  - state=Waiting
+    - message='Failed to apply default image tag "BUSYBOX": couldn''t parse image
+            reference "BUSYBOX": invalid reference format: repository name must be
+            lowercase'
+    - reason=InvalidImageName
+- Conditions:
+  - Type=Status
+  - Initialized=True
+  - Ready=False
+  - ContainersReady=False
+  - PodScheduled=True
+  - PodReadyToStartContainers=True
+
+##### ErrImageNeverPull
+
+- Reproduction: container with ImagePullPolicy: never but container doesn't exist locally
+- Pod status:
+  - status: Pending
+- ContainerStatus:
+  - state=Waiting
+  - reason=ErrImageNeverPull
+- Conditions:
+  - Type=Status
+  - Initialized=True
+  - Ready=False
+  - ContainersReady=False
+  - PodScheduled=True
+  - PodReadyToStartContainers=True
+- Container with missing config map
+  - Reproduction: Container with missing config map
+  - Pod status:
+    - status: Pending
+  - ContainerStatus:
+    - state=Waiting
+      - message=configmap "%s" not found
+      - reason=CreateContainerConfigError
+  - Conditions:
+    - Type=Status
+    - Initialized=True
+    - Ready=False
+    - ContainersReady=False
+    - PodScheduled=True
+    - PodReadyToStartContainers=True
+
+##### CreateContainerConfigError - Missing Key Reference
+
+- Reproduction: Container with existing config map but invalid key reference
+- Pod status:
+  - status: Pending
+- ContainerStatus:
+  - state=Waiting
+    - message=couldn't find key special.how in ConfigMap default/special-config
+    - reason=CreateContainerConfigError
+- Conditions:
+  - Type=Status
+  - Initialized=True
+  - Ready=False
+  - ContainersReady=False
+  - PodScheduled=True
+  - PodReadyToStartContainers=True
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+### Risks and Mitigations
+
+The impact will be a new condition that gets set to True if a user has a error during startup of their pod.
```

Comment:
For alpha we would only allow this if the feature is on.  We could make the feature opt-in.  I don't know if we would wnat this behavior as we don't allow people to not set certain conditions.  

I could see a future where maybe we expand with optional conditions so rather than a feature toggle we have a configurable control of conditions.  But not sure if this is in scope or not.